usng4j-impl/src/main/java/org/codice/usng4j/impl/UsngCoordinateImpl.java

```diff
@@ -183,20 +207,44 @@ private static UsngCoordinate parseCoordinateString(
       throw new ParseException(message, 0);
     }
 
+    int captureGroupsCount = m.groupCount();
     int zoneNumber = Integer.parseInt(m.group(1));
     char latitudeBandLetter = m.group(2).toCharArray()[0];
 
     if (m.group(3) != null) {
       char columnLetter = m.group(3).toCharArray()[0];
       char rowLetter = m.group(3).toCharArray()[1];
 
-      if (!(StringUtils.isEmpty(m.group(4)) || StringUtils.isEmpty(m.group(5)))) {
-        int easting = Integer.parseInt(m.group(4).trim());
-        int northing = Integer.parseInt(m.group(5).trim());
+      if (captureGroupsCount > 4
+          && !(StringUtils.isEmpty(m.group(4)) || StringUtils.isEmpty(m.group(5)))) {
+        String easting = m.group(4).trim();
+        String northing = m.group(5).trim();
+        int eastingInt = Integer.parseInt(easting);
+        int northingInt = Integer.parseInt(northing);
+
+        result =
+            new UsngCoordinateImpl(
+                zoneNumber, latitudeBandLetter, columnLetter, rowLetter, eastingInt, northingInt);
+      } else if (!StringUtils.isEmpty(m.group(4))) {
+        // full numerical location with easting and northing values given as n+n digits
+        String numericalLocation = m.group(4).trim();
```

Comment:
If easting length + northing length is not even, would we want to divide easting length + northing length by 2?  It seems like that would be an invalid MGRS.

Reply:
No we wouldn't want to use those easting/northing values because they would be invalid. I chose to add a check that will throw a `ParseException` if the numerical location can't be evenly split.